tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py

```diff
@@ -0,0 +1,175 @@
+import logging
+import pytest
+
+from ocs_ci.ocs import ocp, constants, defaults
+from ocs_ci.ocs.cluster import (
+    is_flexible_scaling_enabled,
+    check_ceph_health_after_add_capacity,
+)
+from ocs_ci.framework.testlib import tier4c, E2ETest, ignore_leftovers
+from ocs_ci.framework import config
+from ocs_ci.ocs.resources.pod import (
+    get_all_pods,
+    wait_for_pods_to_be_running,
+)
+from ocs_ci.ocs.node import (
+    taint_nodes,
+    untaint_nodes,
+    get_all_nodes,
+    check_taint_on_nodes,
+    get_node_objs,
+)
+from ocs_ci.ocs.resources import storage_cluster
+from ocs_ci.framework.pytest_customization.marks import bugzilla
+from ocs_ci.helpers.sanity_helpers import Sanity
+
+
+logger = logging.getLogger(__name__)
+
+
+@tier4c
+@ignore_leftovers
+@bugzilla("1992472")
+@pytest.mark.polarion_id("OCS-2705")
+class TestNonOCSTaintAndTolerations(E2ETest):
+    """
+    Test to test non ocs taints on ocs nodes
+    and toleration
+    """
+
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+
+        """
+        self.sanity_helpers = Sanity()
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        """
+        Make sure all nodes are untainted
+
+        """
+
+        def finalizer():
+            nodes = get_node_objs(node_names=get_all_nodes())
+            assert untaint_nodes(
+                taint_label="xyz=true:NoSchedule", nodes_to_untaint=nodes
+            ), "Failed to untaint"
+
+        request.addfinalizer(finalizer)
+
+    def test_non_ocs_taint_and_tolerations(self):
+        """
+        Test runs the following steps
+        1. Check if nodes are not ocs tainted
+        2. Taint ocs nodes with non-ocs taint
+        3. Set tolerations on storagecluster, subscription and configmap
+        4. Add Capacity
+        5. Respin all ocs pods and check if it runs on ocs nodes with tolerations
+
+        """
+
+        # Check ocs taints on nodes
+        assert not check_taint_on_nodes(), "Nodes already ocs tainted"
```

Comment:
Do we need to fail here, since the actual test case is not run? Instead, we can skip the test if nodes are OCS tainted.